pkg/controller/context.go

```diff
@@ -41,4 +42,6 @@ type Context struct {
 	DefaultIssuerKind                  string
 	DefaultACMEIssuerChallengeType     string
 	DefaultACMEIssuerDNS01ProviderName string
+	// Metrics
+	Metrics *metrics.Metrics
```

Comment:
If we want to instrument things that don't have access to this context, things will probably get painful.

Can we remove this field for now, and create a package level var in metrics (e.g. `metrics.Default`) that is referenced everywhere?

This should make it easier to reference the metrics collector from anywhere in the codebase.